docs/distributed_write.rst

```diff
@@ -0,0 +1,234 @@
+    Worker 2:  [FragmentMetadata(id=0, files=...)]
+
+
+Now, use :meth:`lance.fragment.FragmentMetadata.to_json` to serialize the fragment metadata,
+and collect all serialized metadata on a single worker to execute the final commit operation.
+
+.. testsetup::
+
+    from lance.fragment import write_fragments
+
+    shutil.rmtree("./dist_write", ignore_errors=True)
+    data_uri = "./dist_write"
+    schema = pa.schema([
+        ("a", pa.int32()),
+        ("b", pa.string()),
+    ])
+
+    data1 = {
+        "a": [1, 2, 3],
+        "b": ["x", "y", "z"],
+    }
+    fragments_1 = write_fragments(data1, data_uri, schema=schema)
+    data2 = {
+        "a": [4, 5, 6],
+        "b": ["u", "v", "w"],
+    }
+    fragments_2 = write_fragments(data2, data_uri, schema=schema)
+
+.. testcode:: new_data
+
+    import json
+    from lance import FragmentMetadata, LanceOperation
+
+    # Serialize Fragments into JSON data
+    fragments_json1 = [json.dumps(fragment.to_json()) for fragment in fragments_1]
+    fragments_json2 = [json.dumps(fragment.to_json()) for fragment in fragments_2]
+
+    # On one worker, collect all fragments
+    all_fragments = [FragmentMetadata.from_json(f) for f in \
+        fragments_json1 + fragments_json2]
+
+    # Commit the fragments into a single dataset
+    # Use LanceOperation.Overwrite to overwrite the dataset or create new dataset.
+    op = lance.LanceOperation.Overwrite(schema, all_fragments)
+    read_version = 0 # Because it is empty at the time.
+    lance.LanceDataset.commit(
+        data_uri,
+        op,
+        read_version=read_version,
+    )
+
+    # We can read the dataset using the Lance API:
+    dataset = lance.dataset(data_uri)
+    assert len(dataset.get_fragments()) == 2
+    assert dataset.version == 1
+    print(dataset.to_table().to_pandas())
+
+.. testoutput:: new_data
+
+        a  b
+     0  1  x
+     1  2  y
+     2  3  z
+     3  4  u
+     4  5  v
+     5  6  w
+
+Append data
+------------
+
+Appending additional data follows a similar process. Use :py:class:`lance.LanceOperation.Append` to commit the new fragments,
+ensuring that the ``read_version`` is set to the current dataset's version.
+
+.. code-block:: python
+    :emphasize-lines: 2,4,5
+
+    ds = lance.dataset(data_uri)
+    read_version = ds.version
+
+    op = lance.LanceOperation.Append(schema, all_fragments)
+    lance.LanceDataset.commit(
+        data_uri,
+        op,
+        read_version=read_version,
+    )
+
+Add New Columns
+---------------
+
+`Lance Format excels at operations such as adding columns <./format.rst>`_.
+Thanks to its two-dimensional layout
+(`see this blog post <https://blog.lancedb.com/designing-a-table-format-for-ml-workloads/>`_),
+adding new columns is highly efficient since it avoids copying the existing data files.
+Instead, the process involves simply create new data files and linking it to the existing dataset
+using metadata-only operations.
```

Comment:
s/linking it/linking them

Reply:
Maybe, "Instead, the process simply creates new data files and links them to the existing dataset using metadata-only operations."